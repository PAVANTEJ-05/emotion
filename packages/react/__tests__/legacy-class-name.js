/** @jsx jsx */
import 'test-utils/setup-env'
import { jsx, css } from '@emotion/react'
import { render } from '@testing-library/react'

test('string className', () => {
  const { container } = render(
    <div
      className="legacy__class"
      css={css`
        color: hotpink;
      `}
    />
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('object className', () => {
  const { container } = render(
    <div
      className={{ toString: () => 'legacy__class' }}
      css={css`
        color: hotpink;
      `}
    />
  )

  expect(container.firstChild).toMatchSnapshot()
})
