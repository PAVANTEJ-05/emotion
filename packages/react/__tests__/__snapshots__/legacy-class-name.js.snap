// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object className 1`] = `
.emotion-0 {
  color: hotpink;
}

<div
  class="legacy__class emotion-0"
/>
`;

exports[`string className 1`] = `
.emotion-0 {
  color: hotpink;
}

<div
  class="legacy__class emotion-0"
/>
`;
